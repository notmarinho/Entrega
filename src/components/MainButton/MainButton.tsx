import {Pressable, StyleSheet, Text, PressableProps} from 'react-native';
import React, {FC} from 'react';

interface MainButtonProps extends PressableProps {
  title: string;
}

const MainButton: FC<MainButtonProps> = ({title, ...props}) => {
  return (
    <Pressable style={styles.container} {...props}>
      <Text style={styles.label}>{title}</Text>
    </Pressable>
  );
};

export default MainButton;

const styles = StyleSheet.create({
  container: {
    width: '100%',
    height: 50,
    backgroundColor: '#303030',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 5,
  },
  label: {
    color: '#fff',
  },
});
