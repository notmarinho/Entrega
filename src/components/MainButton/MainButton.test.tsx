// MainButton Test
import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import MainButton from './MainButton';
import {ReactTestInstance} from 'react-test-renderer';

describe('MainButton', () => {
  it('should render correctly', () => {
    const {toJSON} = render(<MainButton title="This is my Title" />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call onPress when pressed', () => {
    const onPress = jest.fn();
    const {getByTestId} = render(
      <MainButton title="This is my Title" onPress={onPress} />,
    );
    fireEvent.press(getByTestId('MainButton'));
    expect(onPress).toHaveBeenCalled();
  });

  it('should render the title correctly', () => {
    const {getByTestId} = render(
      <MainButton title="This is my Title" onPress={jest.fn()} />,
    );
    const button = getByTestId('MainButton');
    const label = button.children[0] as ReactTestInstance;

    expect(label.props.children).toBe('This is my Title');
  });
});
