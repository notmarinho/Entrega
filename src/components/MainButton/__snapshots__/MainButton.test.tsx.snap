// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainButton should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 10,
      "height": 50,
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="MainButton"
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 18,
      }
    }
  >
    This is my Title
  </Text>
</View>
`;
